security/management/commands/purge_logs.py

```diff
@@ -0,0 +1,144 @@
+        'celery': CeleryTaskLog
+    }
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument('--type', action='store', dest='type', default='input',
+                            help='Tells Django what type of requests should be removed '
+                                 '({}}).'.format('/'.join(self.models.keys())))
+        parser.add_argument('--expiration', action='store', dest='expiration',
+                            help='Sets the timedelta from which logs will be removed.', required=True)
+        parser.add_argument('--noinput', action='store_false', dest='interactive', default=True,
+                            help='Tells Django to NOT prompt the user for input of any kind.')
+        parser.add_argument('--backup', action='store', dest='backup', default=False,
+                            help='Tells Django where to backup removed logs.')
+
+    def backup_logs_and_clean_data(self, qs, backup_path):
+        for timestamp in qs.datetimes('created_at', 'day', tzinfo=utc):
+            min_timestamp = datetime.combine(timestamp, time.min).replace(tzinfo=utc)
+            max_timestamp = datetime.combine(timestamp, time.max).replace(tzinfo=utc)
+            qs_filtered_by_day = qs.filter(
+                **{'created_at__range': (min_timestamp, max_timestamp)}
+            )
+            if backup_path:
+                log_file_name = os.path.join(backup_path, str(timestamp.date()))
+
+                if storage.exists('{}.json.gz'.format(log_file_name)):
+                    i = 1
+                    while storage.exists('{}({}).json.gz'.format(log_file_name, i)):
+                        i += 1
+                    log_file_name = '{}({})'.format(log_file_name, i)
+
+                self.stdout.write(4 * ' ' + 'generate backup file: {}.json.gz'.format(log_file_name))
+
+                with storage.open('{}.json.gz'.format(log_file_name), 'wb') as f:
+                    with TextIOWrapper(gzip.GzipFile(filename='{}.json'.format(log_file_name),
+                                                     fileobj=f, mode='wb')) as gzf:
+                        json.dump(
+                            lazy_serialize_qs_without_pk(qs_filtered_by_day), gzf, cls=DjangoJSONEncoder, indent=5
+                        )
+            self.stdout.write(4 * ' ' + 'clean logs for day {}'.format(timestamp.date()))
+            qs_filtered_by_day.delete()
+
+    def handle(self, expiration, type, **options):
+        # Check we have the correct values
+        unit = expiration[-1]
+        amount = expiration[0:-1]
+        try:
+            amount = int(amount)
+        except ValueError:
+            raise CommandError('Invalid expiration format')
+
+        if unit not in UNIT_OPTIONS:
+            raise CommandError('Invalid expiration format')
+
+        model = self.get_model(type)
+        qs = model.objects.filter(**{'created_at__lte': UNIT_OPTIONS[unit](amount)})
```

Comment:
Unnecessary dict comprehension. :)

Reply:
yes it was refactorization. I will fix it